crucible-saw/src/Lang/Crucible/Backend/SAWCore.hs

```diff
@@ -856,7 +863,7 @@ evaluateExpr sym sc cache = f []
             B.OrderedSemiRingNatRepr     -> join (scNatLe sc <$> eval env xe <*> eval env ye)
 
         B.NotPred x ->
-          SAWExpr <$> (SC.scNot sc =<< f env x)
+          goNeg env x
 
         B.ConjPred xs ->
```

Comment:
Should we be using `goNeg` in the `Negative` case below?

Reply:
Ah, yes, of course. I'll take care of it.

Reply:
Fixed in 6d6e40460e0fcc0027f5e931cc90b1f05b89e3bd.